elements/src/components/pos-add-new-thing/pos-add-new-thing.tsx

```diff
@@ -14,6 +14,11 @@ export class PosAddNewThing {
     this.dialog.showModal();
   }
 
+  @Listen('pod-os:link')
```

Comment:
This gets emitted on submit, so seems like an easy solution for this case.
I haven't been able to think of a generic solution for closing the dialog on submit given that the contents of the dialog are not within its control.